src/lib/paste.ts

```diff
@@ -85,6 +87,23 @@ export const handlePaste = (
   const parent = findParentNode((node) => node.type === schema.nodes.section)(
     tr.selection
   )
+
+  if (
+    slice.content.firstChild &&
+    (slice.content.firstChild.type == schema.nodes.table_row ||
+      slice.content.firstChild.type == schema.nodes.table_cell)
+  ) {
+    const newTable = createTableFromSlice(slice, view.state.schema)
+    if (newTable) {
+      const tr = view.state.tr.insert(
+        view.state.selection.$anchor.pos,
+        newTable
+      )
+      view.dispatch(tr)
+      return true
+    }
+  }
```

Comment:
Issue of copying cells or rows related to the structure of [wrapMap](https://github.com/ProseMirror/prosemirror-view/blob/0c3e5d35baf8af161dd2ed7b2e566c75362abfbb/src/clipboard.ts#L197-L199) as it assumes will be `table -> tbody` but in our case that will be `figure -> table -> tbody` figure which is [tableElement](https://github.com/Atypon-OpenSource/manuscripts-transform/blob/master/src/schema/nodes/table_element.ts#L41C1-L48C9) node in DOM, so if you change that code locally to add `figure` at first in each array issue will be fixed.

We can fix that from our side by using `transformPastedHTML` in [EditorView](https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/6d24b2031be40663f75b20d2f03d089ca3ac5582/src/configs/ManuscriptsEditor.ts#L82)
```
transformPastedHTML: (html) => {
      // add figure which is table_element node in DOM
      if (html.includes('table')) {
        const doc = new DOMParser().parseFromString(html, 'text/html');
        doc.querySelectorAll('table').forEach((value) => {
          if (value.parentElement?.tagName !== 'figure') {
            value.parentElement?.insertBefore(document.createElement('figure'), value)
          }
        })
        return doc.body.innerHTML
      }
      return html
    }
```